### PR TITLE
[W5.7][F10-4] Lee Yi Can

### PR DESCRIPTION
**Pull request by yican95**
Ignore the first commit. Forgot to branch for the week 4 LOs.

### CODE REVIEW

**Comment by m4dkip** [comment]
Good job. Proper encapsulation used and user guide updated neatly.

**Comment by azhikai** [comment]
Good use of commit to keep track of small changes. 
New feature added along with its unit test.

**Comment by HyperionNKJ** [comment]
Good job! User guide updated well. You may want to extend sort to tags :)

**Comment by rebstan97** [comment]
Nicely done! All tests updated neatly.

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -91,6 +91,11 @@ Returns `John Doe` but not `john`.
 * `find Betsy Tim John` +
 Returns Any person having names `Betsy`, `Tim`, or `John`.
 
+=== Sorting all entries: `sort`
```
Comment (dalessr) [inline docs/UserGuide.adoc:94]: Font size different from the others (as you are using `===`)?

File: src/seedu/addressbook/data/AddressBook.java [hunk 2]
```diff
@@ -61,6 +61,11 @@ public void clear() {
         allPersons.clear();
     }
 
+    /**
+     * Sorts all persons in address book in alphabetical order.
+     */
+    public void sort() { allPersons.sort(); }
```
Comment (dalessr) [inline src/seedu/addressbook/data/AddressBook.java:67]: Coding standard violation here. Split into multiple lines.

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
+package seedu.addressbook.commands;
+
+import static org.junit.Assert.*;
```
Comment (dalessr) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:3]: Coding standard violation here. Should specify all the classes used in this file.

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+import java.util.Collections;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
```
Comment (dalessr) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:8]: Split the import into multiple groups as the packages are imported from different sources.

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        assertSortSuccessful(emptyAddressBook, emptyList, emptyAddressBook);
+
+        // sort on address book
+        assertSortSuccessful(addressBook, listWithEveryone, sortedAddressBook);
+    }
+
+    /**
+     * Creates a new sort command.
+     *
+     * @return
```
Comment (dalessr) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:60]: Description missing here. What does this method return?

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+     */
+    private void assertCommandBehaviour(SortCommand sortCommand, String expectedMessage,
+            AddressBook expectedAddressBook, AddressBook actualAddressBook) {
+
+        CommandResult result = sortCommand.execute();
+
+        assertEquals(expectedMessage, result.getFeedbackToUser());
+        assertEquals(expectedAddressBook.getAllPersons(), actualAddressBook.getAllPersons());
+    }
+}
```
Comment (dalessr) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:95]: New line is required at the end of each file

**Review by dalessr — CHANGES_REQUESTED** [review]
@yican95 Well done for the sort command 👍 This PR is rejected as it is not a clean PR.

Please close the PR after looking at the comments below.

**Pull request closed without merging**